### ISSUE TITLE
Initial setup produces 400 creating network

### ISSUE THREAD

**Issue by exKAjFASH** [OpenVPN/terraform-provider-openvpn-cloud#16]
Trying to use simple configuration to setup fresh environment:
```terraform
terraform {
  required_providers {
    openvpn-cloud = {
      source = "OpenVPN/openvpn-cloud"
      version = "0.0.7"
    }
  }
}

provider "openvpn-cloud" {
  base_url = "https://sometest.api.openvpn.com"
}

resource "openvpncloud_network" "vpn_network" {
  name = "test_network"
  default_connector {
    name = "test_connector"
    vpn_region_id = var.region
  }
  default_route {
    value = "10.1.1.0/24"
    type = "IP_V4"
  }
  provider = openvpn-cloud
}
```
Commands `terraform init` and `terraform plan` works as expected, but `terraform apply` fails:
```
terraform apply  

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # openvpncloud_network.vpn_network will be created
  + resource "openvpncloud_network" "vpn_network" {
      + description     = "Managed by Terraform"
      + egress          = true
      + id              = (known after apply)
      + internet_access = "LOCAL"
      + name            = "test_network"
      + system_subnets  = (known after apply)

      + default_connector {
          + id                = (known after apply)
          + ip_v4_address     = (known after apply)
          + ip_v6_address     = (known after apply)
          + name              = "test_connector"
          + network_item_id   = (known after apply)
          + network_item_type = (known after apply)
          + vpn_region_id     = "us-east-1"
        }

      + default_route {
          + id    = (known after apply)
          + type  = "IP_V4"
          + value = "10.1.1.0/24"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

openvpncloud_network.vpn_network: Creating...
╷
│ Error: Status code: 400, Response body: {"path":"/api/beta/networks","requestId":"faf40205-56","status":400,"statusError":"Bad Request","timestamp":1671744332781}
│ 
│   with openvpncloud_network.vpn_network,
│   on main.tf line 16, in resource "openvpncloud_network" "vpn_network":
│   16: resource "openvpncloud_network" "vpn_network" {
│ 
╵
```
Ui on OpenVPN shows created Connector and Network, but without Routes. Changing `default_route` to use DOMAIN has same error.

**Comment by arslanbekov** [comment]
I try to run this terraform code:
```
terraform {
  required_providers {
    openvpn-cloud = {
      source  = "OpenVPN/openvpn-cloud"
      version = "0.0.7"
    }
  }
}

provider "openvpn-cloud" {
  base_url = "https://mycompany.api.openvpn.com"
}

resource "openvpncloud_network" "vpn_network" {
  name = "test_network"
  default_connector {
    name          = "test_connector"
    vpn_region_id = "fi-hel"
  }
  default_route {
    value = "10.1.1.0/24"
    type  = "IP_V4"
  }
  provider = openvpn-cloud
}
```

And as I see my resource has been created:
```
> terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # openvpncloud_network.vpn_network will be created
  + resource "openvpncloud_network" "vpn_network" {
      + description     = "Managed by Terraform"
      + egress          = true
      + id              = (known after apply)
      + internet_access = "LOCAL"
      + name            = "test_network"
      + system_subnets  = (known after apply)

      + default_connector {
          + id                = (known after apply)
          + ip_v4_address     = (known after apply)
          + ip_v6_address     = (known after apply)
          + name              = "test_connector"
          + network_item_id   = (known after apply)
          + network_item_type = (known after apply)
          + vpn_region_id     = "fi-hel"
        }

      + default_route {
          + id    = (known after apply)
          + type  = "IP_V4"
          + value = "10.1.1.0/24"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

openvpncloud_network.vpn_network: Creating...
openvpncloud_network.vpn_network: Creation complete after 1s [id=c17a238a-cbda-4fbc-a1a1-0a2e6b5009cd]
╷
│ Warning: The default connector for this network needs to be set up manually
│ 
│   with openvpncloud_network.vpn_network,
│   on main.tf line 16, in resource "openvpncloud_network" "vpn_network":
│   16: resource "openvpncloud_network" "vpn_network" {
│ 
│ Terraform only creates the OpenVPN Cloud default connector object for this network, but additional manual steps are required to associate a host in
│ your infrastructure with this connector. Go to https://openvpn.net/cloud-docs/connector/ for more information.
╵

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
<img width="1225" alt="Screenshot 2022-12-26 at 15 05 45" src="https://user-images.githubusercontent.com/579213/209542022-61bc75ee-2cb5-4123-bfab-379622bd02b3.png">

**Comment by arslanbekov** [comment]
I was able to reproduce the issue. It is reproduced if you already have a network. Or you create >1 network at once

**Comment by iamunni** [comment]
edit: ignore this comment. it is working now

@arslanbekov even with the terraform code you shared above, I'm getting error

code:
```
terraform {
  required_providers {
    openvpncloud = {
      source = "OpenVPN/openvpn-cloud"
      version = "0.0.7"
    }
  }
}

provider "openvpncloud" {
  base_url = "https://iamunni.api.openvpn.com"
}

resource "openvpncloud_network" "vpn_network" {
  name = "test_network"
  default_connector {
    name          = "test_connector"
    vpn_region_id = "fi-hel"
  }
  default_route {
    value = "10.1.1.0/24"
    type  = "IP_V4"
  }
  provider = openvpncloud
}
```
apply output:
```
Terraform will perform the following actions:

  # openvpncloud_network.vpn_network will be created
  + resource "openvpncloud_network" "vpn_network" {
      + description     = "Managed by Terraform"
      + egress          = true
      + id              = (known after apply)
      + internet_access = "LOCAL"
      + name            = "test_network"
      + system_subnets  = (known after apply)

      + default_connector {
          + id                = (known after apply)
          + ip_v4_address     = (known after apply)
          + ip_v6_address     = (known after apply)
          + name              = "test_connector"
          + network_item_id   = (known after apply)
          + network_item_type = (known after apply)
          + vpn_region_id     = "fi-hel"
        }

      + default_route {
          + id    = (known after apply)
          + type  = "IP_V4"
          + value = "10.1.1.0/24"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
openvpncloud_network.vpn_network: Creating...
╷
│ Error: Status code: 400, Response body: {"path":"/api/beta/networks/e531fcfa-b0e3-49a5-a43c-fdee8267aaa1/routes","requestId":"51370d35-110","status":400,"statusError":"Bad Request","timestamp":1677007262877}
│
│   with openvpncloud_network.vpn_network,
│   on main.tf line 19, in resource "openvpncloud_network" "vpn_network":
│   19: resource "openvpncloud_network" "vpn_network" {
│
╵
```

I can see the network created. but the route is not added
![image](https://user-images.githubusercontent.com/62323890/220440689-4dc92412-1b76-472f-8030-13a7c1eb802f.png)
![image](https://user-images.githubusercontent.com/62323890/220440773-479d4970-7500-44b8-bddd-aefc393f8f41.png)


I dont understand what is causing this. am I missing anything obvious? thinking. I'm able to created the network successfully with same config using the swagger UI.

**Comment by iamunni** [comment]
I dont understand what is causing this. am I missing anything obvious? :thinking: 
created the network  successfully with same config using the swagger UI.

**Comment by sotNaz** [comment]
@iamunni I have the exact same issue and I am using the exact same piece of code. What was the fix in your case ?

```
Terraform will perform the following actions:

  # openvpncloud_network.vpn_network will be created
  + resource "openvpncloud_network" "vpn_network" {
      + description     = "Managed by Terraform"
      + egress          = true
      + id              = (known after apply)
      + internet_access = "LOCAL"
      + name            = "test_network"
      + system_subnets  = (known after apply)

      + default_connector {
          + id                = (known after apply)
          + ip_v4_address     = (known after apply)
          + ip_v6_address     = (known after apply)
          + name              = "test_connector"
          + network_item_id   = (known after apply)
          + network_item_type = (known after apply)
          + vpn_region_id     = "eu-central-1"
        }

      + default_route {
          + id    = (known after apply)
          + type  = "IP_V4"
          + value = "10.1.1.0/24"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions in workspace "openvpn-cloud"?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

openvpncloud_network.vpn_network: Creating...
╷
│ Error: Status code: 400, Response body: {"path":"/api/beta/networks/8b0e7bef-1af5-4f6e-b12a-837223808885/routes","requestId":"e1918ca9-328","status":400,"statusError":"Bad Request","timestamp":1677076130439}
│ 
│   with openvpncloud_network.vpn_network,
│   on openvpn.tf line 17, in resource "openvpncloud_network" "vpn_network":
│   17: resource "openvpncloud_network" "vpn_network" {
│ 
╵
```

**Comment by vladimir-kozyrev** [comment]
@exKAjFASH  @iamunni @sotNaz could you check whether the new release fixes the issue?

**Comment by Rycieos** [comment]
@vladimir-kozyrev I just tested v0.0.8, and it worked exactly like v0.0.7; still not working.

**Comment by vladimir-kozyrev** [comment]
@Rycieos hmm... I can reproduce this with 0.0.7, but not with 0.0.8.
This is the code that I am using.

```terraform
resource "openvpncloud_network" "vpn_network" {
  name = "test_network"
  default_connector {
    name          = "test_connector"
    vpn_region_id = "fi-hel"
  }
  default_route {
    value = "10.1.1.0/24"
    type  = "IP_V4"
  }
}
```

**Comment by Rycieos** [comment]
@vladimir-kozyrev my mistake. I don't know what I did wrong previously, so I can't report on that. But it is working now:
```terraform
resource "openvpncloud_network" "network" {
  provider = openvpn-cloud
  name     = "network"

  internet_access = "LOCAL"
  egress          = false

  default_connector {
    name          = "my-connector"
    vpn_region_id = "us-pdx"
  }

  default_route {
    type  = "DOMAIN"
    value = "example.com"

    # Bug in the provider tries to set this, but the API doesn't allow default
    # route to have a description.
    description = ""
  }
}
```
Note that if you don't specify a `description` for the `default_route`, the apply will attempt to set one, which will appear to succeed, but will actually not set anything, so each apply will again attempt to set it again. Setting `description = ""` will work around this.

**Comment by vladimir-kozyrev** [comment]
@Rycieos the description should be properly persisted now since the release of `v0.0.9`.

**Comment by Rycieos** [comment]
@vladimir-kozyrev nice, you are correct, it works. Thanks!

**Comment by Rycieos** [comment]
This is no longer working, neither with `v0.0.9` nor `v0.0.10`. I'm assuming it's an API change, since it used to work. Most likely the API no longer accepts default_route info, since the GUI doesn't have any information on it.

> Error: Status code: 400, Response body: {"path":"/api/beta/networks","requestId":"cb03893f-177","status":400,"statusError":"Bad Request","timestamp":1687800423646}

To be completely clear, the above issue happens with both creating a new resource, and diffing a previously created resource. In the later case, the following plan is shown:
```
  # openvpncloud_network.example will be updated in-place
  ~ resource "openvpncloud_network" "example" {
        id              = "xxxxxxxx-xxxx-xxxx-xxxx-74fd9791f064"
        name            = "Example"
        # (4 unchanged attributes hidden)

      + default_route {
          + description = "Managed by Terraform."
          + type        = "DOMAIN"
          + value       = "example.com"
        }

        # (1 unchanged block hidden)
    }
```
And when applied, the above 400 error occurs.

**Comment by Rycieos** [comment]
This appears to be fixed in `v0.0.11`.

**Comment by sahaqaa** [comment]
We've published new version of Terraform Provider to Registry: https://registry.terraform.io/providers/OpenVPN/cloudconnexa/latest

Going forward only https://registry.terraform.io/providers/OpenVPN/cloudconnexa/latest will be developed.

Please give new provider a try, hopefully it will fix your issue.

In case if problem persists - please open new "Issue" (https://github.com/OpenVPN/terraform-provider-cloudconnexa/issues)

Sorry for inconvenience.
